event-driven-architecture/src/main/java/com/iluwatar/eda/framework/EventDispatcher.java

```diff
@@ -22,19 +22,16 @@
  */
 package com.iluwatar.eda.framework;
 
-import com.iluwatar.eda.event.Event;
-
 import java.util.HashMap;
 import java.util.Map;
 
 /**
  * Handles the routing of {@link Event} messages to associated handlers.
  * A {@link HashMap} is used to store the association between events and their respective handlers.
- *
  */
 public class EventDispatcher {
 
-  private Map<Class<? extends Event>, Handler<?>> handlers;
+  private Map<Class<? extends Event>, Handler<? extends Event>> handlers;
```

Comment:
This map declaration and the `registerHandler()` signature below, both ensure only 'type-safe' combination of event type and handler can be registered.